rust/cardano-blockchain-types/src/txn_witness.rs

```diff
@@ -0,0 +1,143 @@
+//! Transaction Witness
+use std::fmt::{Display, Formatter};
+
+use dashmap::{DashMap, DashSet};
+use ed25519_dalek::VerifyingKey;
+use pallas::ledger::traverse::MultiEraTx;
+
+use crate::{conversion::vkey_from_bytes, hashes::Blake2b224Hash, TxnIndex};
+
+/// Hash of a witness verifying public key
+pub type VKeyHash = Blake2b224Hash;
+
+/// `WitnessMap` type of `DashMap` with
+/// key as [u8; 28] = (`blake2b_244` hash of the public key)
+/// value as `(Bytes, Vec<u8>) = (public key, tx index within the block)`
+type WitnessMap = DashMap<VKeyHash, (VerifyingKey, DashSet<TxnIndex>)>;
+
+#[derive(Debug)]
+/// `TxnWitness` struct to store the witness data.
+pub struct TxnWitness(WitnessMap);
+
+impl TxnWitness {
+    /// Create a new `TxnWitness` from a list of `MultiEraTx`.
+    ///
+    /// # Errors
+    ///
+    /// Only if the witness map does not contain a valid ED25519 public key
```

Comment:
As far as I can see, we can also return the "Unsupported transaction type" error.

Reply:
By the way, this is a reason why I think that this lint is useless. It can (and will) become obsolete very quickly. Both in the documentation and the code itself we can see the return value anyway. Also by choosing `anyhow::Error` we kind of say "I don't care about the exact error". But this isn't a hill I want to die on. 🙃